src/AI.pm

```diff
@@ -82,8 +82,22 @@ use constant MANUAL => 1;
 # AI is turned on.
 use constant AUTO => 2;
 
+my $AI = AUTO;
```

Comment:
This is pretty aggressive in preventing unauthorized state changes. I'd recommend using `our` instead of `my`, in case somebody has a legitimate reason to bypass `AI::state`. Maybe with a comment saying "use AI::state instead of modifying this variable directly".